django_cte/query.py

```diff
@@ -75,9 +79,42 @@ def generate_sql(cls, connection, query, as_sql):
         for cte in query._with_ctes:
             if django.VERSION > (4, 2):
                 _ignore_with_col_aliases(cte.query)
-            compiler = cte.query.get_compiler(connection=connection)
+
+            alias = query.alias_map.get(cte.name)
+            is_left_outer = (
+                isinstance(alias, QJoin) and alias.join_type == LOUTER
+            )
+
+            if django.VERSION >= (4, 0):
+                compiler = cte.query.get_compiler(
+                    connection=connection, elide_empty=not is_left_outer
```

Comment:
What do you think of making `is_left_outer` more generic/applicable to this context?
```suggestion
            should_elide_empty = (
                not isinstance(alias, QJoin) or alias.join_type != LOUTER
            )

            if django.VERSION >= (4, 0):
                compiler = cte.query.get_compiler(
                    connection=connection, elide_empty=should_elide_empty
```